src/gemini_adapter/src/gemini_adapter/_impl.py

```
@@ -0,0 +1,100 @@
+        """Initialize the adapter.
+
+        Args:
+            base_url: Base URL of the Gemini FastAPI service.
+                     Defaults to localhost:8000.
+
+        """
+        self.client = GeminiHTTPClient(base_url=base_url)
+
+    def send_message(self, user_id: str, message: str) -> str:
+        """Send a message via the Gemini FastAPI service.
+
+        Args:
+            user_id: Unique identifier for the user.
+            message: The message text to send.
+
+        Returns:
+            The AI's response as a string.
+
+        Raises:
+            ValueError: If user_id or message is empty.
+
+        """
+        if not user_id:
+            msg = "user_id cannot be empty"
+            raise ValueError(msg)
+        if not message:
+            msg = "message cannot be empty"
+            raise ValueError(msg)
+
+        response = self.client.send_message_chat_post(
+            json_body={"user_id": user_id, "message": message},
+        )
+        return response.response
+
+    def get_conversation_history(self, user_id: str) -> list[Message]:
+        """Retrieve conversation history via the Gemini FastAPI service.
+
+        Args:
+            user_id: Unique identifier for the user.
+
+        Returns:
+            A list of Message objects representing the conversation history.
+
+        Raises:
+            ValueError: If user_id is empty.
+
+        """
+        if not user_id:
+            msg = "user_id cannot be empty"
+            raise ValueError(msg)
+
+        response = self.client.get_conversation_history_history_user_id_get(
+            user_id=user_id,
+        )
+        if hasattr(response, "messages") and response.messages:
+            return [
+                Message(
+                    role=msg.role,
+                    content=msg.content,
+                )
+                for msg in response.messages
+            ]
+        return []
+
+    def clear_conversation(self, user_id: str) -> bool:
```

For the clear_conversation() method, you should raise an error on unexpected responses so messagers don't assume success when given an ambiguous or faulty result